### ISSUE TITLE
[BUG] Compilation error

### ISSUE THREAD

**Issue by anthonymoretti** [zeshuaro/google_api_headers#558]
### Steps to Reproduce

Build for Android.

### Expected behaviour

App should compile.

### Actual behaviour

App doesn't compile and this is the error:

> e: file:///_/.pub-cache/hosted/pub.dev/google_api_headers-4.3.1/android/src/main/kotlin/io/github/zeshuaro/google_api_headers/GoogleApiHeadersPlugin.kt:44:34 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type SigningInfo?
> e: file:///_/.pub-cache/hosted/pub.dev/google_api_headers-4.3.1/android/src/main/kotlin/io/github/zeshuaro/google_api_headers/GoogleApiHeadersPlugin.kt:55:33 Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Array<(out) Signature!>?
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':google_api_headers:compileDebugKotlin'.
> > A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
>    > Compilation error. See log for more details

I can make it run by adding 3 null checks in this block of code in GoogleApiHeadersPlugin.kt:

```
    @SuppressLint("PackageManagerGetSignatures")
    @UiThread
    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method == "getSigningCertSha1") {
            try {
                val packageManager = context!!.packageManager
                val args = call.arguments<String>()!!
                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
                    packageManager.getPackageInfo(
                        args,
                        PackageManager.GET_SIGNING_CERTIFICATES
                    ).signingInfo?.apkContentsSigners?.forEach { signature ->
                        parseSignature(
                            signature,
                            result
                        )
                    }
                } else {
                    @Suppress("DEPRECATION")
                    packageManager.getPackageInfo(
                        args,
                        PackageManager.GET_SIGNATURES
                    ).signatures?.forEach { signature -> parseSignature(signature, result) }
                }

            } catch (e: Exception) {
                result.error("ERROR", e.toString(), null)
            }
        } else {
            result.notImplemented()
        }
    }
```

### Flutter doctor output

[✓] Flutter (Channel stable, 3.27.1, on macOS 15.2 24C101 darwin-arm64, locale en-US)
    • Flutter version 3.27.1 on channel stable at /_/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 17025dd882 (7 days ago), 2024-12-17 03:23:09 +0900
    • Engine revision cb4b5fff73
    • Dart version 3.6.0
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /_/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /_/Library/Java/JavaVirtualMachines/jbr-17.0.12/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment JBR-17.0.12+1-1207.37-nomod (build 17.0.12+1-b1207.37)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

**Comment by anthonymoretti** [comment]
Also present in 4.3.2. Updating the package meant I had to repeat the workaround again. Is it possible for you to add those 3 null checks?

**Comment by zeshuaro** [comment]
Can you share the exact steps to reproduce the issues? I tried running `flutter build appbundle` but not seeing the same errors you shared

**Comment by anthonymoretti** [comment]
1. Open project in VS Code.
2. Select an Android emulator as the current device.
3. In "Run and Debug", click the play button "Start Debugging".
4. Observe the error in the Debug Console.

This is happening on 4.3.3 as well.

I don't know if any of this info helps, but this is what I'm using:

Android Studio Ladybug Feature Drop | 2024.2.2

app/build.gradle:
```
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_17
    }
```

settings.gradle:
```
plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version '8.8.0' apply false
    id "org.jetbrains.kotlin.android" version "1.8.22" apply false
}
```

In the google_api_headers build.gradle it has:
```
    kotlinOptions {
        jvmTarget = '1.8'
    }
```

**Comment by zeshuaro** [comment]
I've tried to create a brand-new Flutter app with `flutter create` and added the dependency, but still unable to reproduce the issue you're facing. Can you share a minimal reproducible example?

**Comment by anthonymoretti** [comment]
TL;DR: Is it possible the problem is caused because the package targets Java version 8?

--

I created a brand new project and made these changes to match my existing project:

app/build.gradle:
```
ndkVersion = "27.0.12077973"

compileOptions {
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}

kotlinOptions {
    jvmTarget = JavaVersion.VERSION_17
}

minSdk = 24

ndk {
    abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86_64'
}
```

settings.gradle:

I used the Upgrade Assistant in Android Studio to upgrade the Android Gradle Plugin from 8.1.0 to 8.8.0:
`id "com.android.application" version '8.8.0' apply false`

In Android Studio in Settings I changed the Gradle JDK from jbr-21 to jbr-17.

Then, in build.gradle, I added this code and this is what creates the problem. It's taken from this issue comment:
https://github.com/flutter/flutter/issues/125181#issuecomment-1555034383
```
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"

    // Taken from:
    // https://github.com/flutter/flutter/issues/125181#issuecomment-1555034383
    afterEvaluate {
        // Check if android block is available.
        if (it.hasProperty('android')) {
            if (it.android.namespace == null) {
                def manifest = new groovy.xml.XmlSlurper().parse(file(it.android.sourceSets.main.manifest.srcFile))
                def packageName = manifest.@package.text()
                // println("Setting ${packageName} as android namespace")
                android.namespace = packageName
            }

            def javaVersion = JavaVersion.VERSION_17
            android {
                def androidApiVersion = 35
                compileSdkVersion androidApiVersion
                defaultConfig {
                    targetSdkVersion androidApiVersion
                }
                compileOptions {
                    sourceCompatibility javaVersion
                    targetCompatibility javaVersion
                }
                tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
                    kotlinOptions {
                        jvmTarget = javaVersion.toString()
                    }
                }
                // println("Setting java version to ${javaVersion.toString()} which is $javaVersion")
                // println("Setting compileSdkVersion and targetSdkVersion to $androidApiVersion")
            }
        }
    }
}
```

I don't know, but if the package were to target Java version 17, and if those 3 null checks from my workaround are added to the package, would it solve the problem? If not, I'll just keep adding the null checks every time I upgrade the package.

**Comment by zeshuaro** [comment]
Were you experiencing the issue described in https://github.com/flutter/flutter/issues/125181, so you had to add that block of code? Because I followed your steps up to and before adding that block of code, and was able to build and run the app successfully. It also seems like that issue is related to an older Flutter version and may not be relevant anymore?

**Comment by anthonymoretti** [comment]
That's right, I just commented out that block of code and tried building for Android and this is error:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':file_previewer'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file: /_/.pub-cache/hosted/pub.dev/file_previewer-0.2.0/android/build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.
```

I could be wrong, but isn't Java 17 the recommended version for Flutter now?

**Comment by zeshuaro** [comment]
So it seems like the problem is coming from the `file_previewer` package you're using where they're not specifying the `namespace` attribute. So you might want to raise an issue for them and make a fix like this: https://github.com/zeshuaro/google_api_headers/pull/121

**Comment by anthonymoretti** [comment]
Thank you, I just shared your suggestion in the relevant [issue](https://github.com/DutchCodingCompany/file_preview/issues/15) on that package.
